Task #150380 feat: Create the search functionality for find joomla user

Pratikshakhandagale opened the pull request.

Pratikshakhandagale committed:
Task #147727 feat: [User Plugin] - Get list of language which are installed on site

Pratikshakhandagale committed:
Task #147727 feat: [Modify code][User Plugin] - Get list of language which are installed on site

thite-amol committed:
Merge pull request #24 from Pratikshakhandagale/147727

Get the a list of language which is installed on-site

Pratikshakhandagale committed:
Task #150380 feat: Create the search functionality for find joomla user

manojLondhe commented on src/users/users.php:
```diff
@@ -368,8 +368,22 @@ public function get()
 			$users = $model->getItems();
 
+			if (!$users)
+			{
```
@Pratikshakhandagale @thite-amol This is a valid request, right?
Shall this be code 200, with message as `No matching users` ?

manojLondhe replied:
I meant return regular response with no users, and a message?

Pratikshakhandagale committed:
Task #150380 feat: Code improvement

Pratikshakhandagale committed:
Task #150380 refactor : Get user data response body change

Response body:

{
    "err_msg": "",
    "err_code": "",
    "response_id": "5e296856caf65",
    "api": "users.users",
    "version": "",
    "data": {
        "result": [],
        "empty_message": "User not found"
    }
}

Pratikshakhandagale committed:
Task #150380 feat: Response format changes

manojLondhe commented on src/users/users.php:
```diff
@@ -352,30 +352,40 @@ public function get()
 	{
```
for getting single user we should use different resource (resource=user)
https://github.com/techjoomla/plg_api_users/blob/95b2d686206a4d460a282c3a82255239e4801400/src/users/user.php#L155

Pratikshakhandagale committed:
Task #150380 feat: Remove Duplicate code
